reactioncommerce/reaction
# Demeteorize file collections and add jobs to convert images

Resolves [reaction-file-collections#21 ](https://github.com/reactioncommerce/reaction-file-collections/issues/21)
Impact: **breaking**  
Type: **refactor**

## Issue
_(WIP until https://github.com/reactioncommerce/reaction-file-collections/pull/28 has been merged)_

This PR implements usage of the job queue for image conversion like discussed in https://github.com/reactioncommerce/reaction-file-collections/issues/21.

**Further it includes new versions of `TempFileStoreWorker` and `RemoteUrlWorker` that don't rely on Meteor anymore.**
Note that I included `TempFileStoreWorker` and `RemoteUrlWorker` in this PR for testing purpouses.

A complementary PR for these two will be submitted to [reaction-file-collections](https://github.com/reactioncommerce/reaction-file-collections).

As soon as that has been merged I will update this PR to remove them and reference from `reaction-file-collections` instead.

## Breaking changes
Removed `meteorFileCollectionsStartup` and refactored the creation of the TempFileStoreWorker and RemoteUrlWorker instances to `setUpFileCollections` of the  no-meteor startup function.


## Testing
1. Add an Image through the Operator UI

2. Make sure that this image is created properly.

3. Check the jobs collection for a completed job of type "convertImage/local":
`db.getCollection('Jobs').find({type: "convertImage/local"})`


4. Add a remote image record to the `cfs.Media.filerecord` collection to be fetched by RemoteUrlWorker

_Example:_
```
{
    "_id" : "MvYtxzTm2Z3JxQrb2",
    "original" : {
        "name" : "abc",
        "type" : [ 
            "image/png"
        ],
        "size" : 1132,
        "updatedAt" : ISODate("1970-01-01T00:00:00.000Z"),
        "remoteURL" : "https://via.placeholder.com/300/09f/abc.png"
    },
    "metadata" : {
        "ownerId" : "czmGeRT8wb94knvRx",
        "productId" : "BCTMZ6HTxFSppJESk",
        "shopId" : "J8Bhq3uTtdgwZx3rz",
        "priority" : 1,
        "toGrid" : 1,
        "workflow" : "new"
    }
}
```

5. Make sure that this image is created properly. (The id provided here should match the basic product created from the fixtures, else lookup the url though the document in the filerecord collection)

6. Check the jobs collection for a completed job of type "convertImage/remote":
`db.getCollection('Jobs').find({type: "convertImage/remote"})`
